app/dashboard/controller.js

```diff
@@ -0,0 +1,118 @@
+import { computed } from '@ember/object';
+import { A } from '@ember/array';
+import { inject as service } from '@ember/service';
+import Controller from '@ember/controller';
+import { task, timeout } from 'ember-concurrency';
+
+const popularNode = '57tnq';
+const noteworthyNode = 'z3sg2';
+
+export default Controller.extend({
+    currentUser: service(),
+
+    init() {
+        this.get('initialLoad').perform();
+        this.get('store').findAll('institution').then(institutions => this.set('institutions', institutions));
+        this.get('getPopularAndNoteworthy').perform(popularNode, 'popular');
+        this.get('getPopularAndNoteworthy').perform(noteworthyNode, 'noteworthy');
+    },
+
+    loading: false,
+    loadingMore: false,
+    initialLoading: true,
+    curPage: 1,
+    filter: '',
+    sortBy: 'last_logged',
+    sortOrder: 'desc',
+    modalOpen: false,
+    nodes: A([]),
+    institutions: A([]),
+    popular: A([]),
+    noteworthy: A([]),
+
+    hasMore: computed('nodes.length', function() {
+        if (this.get('nodes.length') < 10) {
+            return false;
+        }
+        return this.get('totalPages') ? this.get('totalPages') > this.get('curPage') : false;
+    }),
+
+    actions: {
+        more() {
+            this.incrementProperty('curPage');
+            this.get('findNodes').perform(true);
+        },
+        sort(by, order) {
+            this.set('sortBy', by);
+            this.set('sortOrder', order);
+            this.get('findNodes').perform();
+        },
+        setSortBy(by) {
+            this.set('sortBy', by);
+            this.get('findNodes').perform();
+        },
+        setSortOrder(order) {
+            this.set('sortOrder', order);
+            this.get('findNodes').perform();
+        },
+        toggleModal() {
+            this.toggleProperty('modalOpen');
+        },
+        reloadNodes() {
+            this.set('loading', true);
+            this.set('curPage', 1);
+            this.get('initialLoad').perform();
+        },
+    },
+
+    initialLoad: task(function* () {
+        const user = yield this.get('currentUser.user');
+        if (!user) { return; }
+        const nodes = yield user.queryHasMany('nodes', { embed: 'contributors' });
+        this.set('nodes', nodes.slice());
+        this.set('totalNodes', nodes.meta.total);
+        const pages = Math.ceil(nodes.meta.total / nodes.meta.per_page);
+        this.set('totalPages', pages);
+        this.set('initialLoading', false);
```

Comment:
Instead of `initialLoading`, use `initialLoad.isRunning`

Reply:
It'd be nice to use more e-concurrency task state for things like this (instead of `loading`, `loadingMore`, etc.), but the way these tasks are set up doesn't map cleanly and a better way isn't immediately obvious to me. \*shrug*